src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1281,9 +1279,16 @@ internal static string GetWindowsPowerShellModulePath()
         private static string SetModulePath()
         {
             string currentModulePath = GetExpandedEnvironmentVariable(Constants.PSModulePathEnvVar, EnvironmentVariableTarget.Process);
+#if !UNIX
+            // if the current process and user env vars are the same, it means we need to append the machine one as it's incomplete
+            // otherwise, the user modified it and we should use the process one
+            if (string.CompareOrdinal(GetExpandedEnvironmentVariable(Constants.PSModulePathEnvVar, EnvironmentVariableTarget.User), currentModulePath) == 0)
+            {
+                currentModulePath += Path.PathSeparator + GetExpandedEnvironmentVariable(Constants.PSModulePathEnvVar, EnvironmentVariableTarget.Machine);
```

Comment:
Core team recommends the pattern to reduce allocations (compiler optimization using string.Concat with ReadonlySpan<char>)
```suggestion
                currentModulePath = currentModulePath + Path.PathSeparator + GetExpandedEnvironmentVariable(Constants.PSModulePathEnvVar, EnvironmentVariableTarget.Machine);
```

The logic is not clear for me. If we do the "append" and below write the new value to env then we will be in "the user modified it" state.

Reply:
I tried to explain this in the description.  If the User PSModulePath = 'foo' and Machine PSModulePath = 'bar'.  Then a new process will have PSModulePath = 'foo'.  So if the Process PSModulePath is exactly the same as the User PSModulePath, we can say it wasn't modified so we need to append the Machine PSModulePath to make it complete (copying behavior of PATH env var).  I believe this was also WinPS behavior.  However, if the user modifies the Process PSModulePath to 'foo;new' or even just 'new', then we know it isn't the default User PSModulePath (which is still just 'foo') and thus leave it as-is respecting the user's change.